### PR TITLE
Fix tests for Chaplin#780

### PR DESCRIPTION
**Pull request by Florian-R**
I'd really like to see gh-780 merged, so i tried to fix the failling tests.

I had to update grunt-mocha due to some inability to install PhantomJS.

I'm not completly happy with this PR, i'll comment in the code.

Also, i didn't try to see if tests are OK without deps, i'd prefer adress eventuals comments before going on with this

### CODE REVIEW

File: src/chaplin/lib/utils.coffee [hunk 1]
```diff
@@ -84,6 +84,12 @@ utils =
   escapeRegExp: (str) ->
     return String(str or '').replace /([.*+?^=!:${}()|[\]\/\\])/g, '\\$1'
 
+  # DOM Helpers
+  # --------------
+
+  matchesSelector: (node, selector) ->
+    matchesSelector = node.matches || node.webkitMatchesSelector || node.mozMatchesSelector || node.msMatchesSelector
+    matchesSelector.call(node, selector)
```
Comment (Florian-R) [inline src/chaplin/lib/utils.coffee]: Not sure DOM manipulation belongs to Chaplin

File: test/spec/collection_view_spec.coffee [hunk 2]
```diff
@@ -412,51 +412,36 @@ define [
...
-        $css = sinon.stub jQuery.prototype, 'css', -> this
         $animate = sinon.stub jQuery.prototype, 'animate', -> this
 
         createCollection()
         collectionView = new AnimatingCollectionView {collection}
 
-        expect($css.callCount).to.be collection.length
-        expect($css).was.calledWith 'opacity', 0
-
         expect($animate.callCount).to.be collection.length
```
Comment (Florian-R) [inline test/spec/collection_view_spec.coffee:420]: Wastn't so sure about all these tests. It felt weird to just test that style.opacity was '0' on all childs views.

File: src/chaplin/lib/utils.coffee (hunk 1 above)
Comment (akre54) [inline src/chaplin/lib/utils.coffee]: Agreed. But maybe we could use something like https://github.com/akre54/Backbone.NativeView/blob/0.3.2/backbone.nativeview.js#L34-L48 that has older browser support and eagerly finds the selector... if we really need it

File: src/chaplin/views/collection_view.coffee [hunk 3]
```diff
@@ -7,7 +7,7 @@ utils = require 'chaplin/lib/utils'
 
 filterChildren = (nodeList, selector) ->
   return nodeList unless selector
-  for node in nodeList when Backbone.utils.matchesSelector node, selector
+  for node in nodeList when utils.matchesSelector node, selector
```
Comment (akre54) [inline src/chaplin/views/collection_view.coffee]: Any reason we cant just replace the `filterChildren` function with el.qsa in-place?

File: src/chaplin/views/collection_view.coffee [hunk 4]
```diff
@@ -24,7 +24,7 @@ startAnimation = (elem, useCssAnimation, cls) ->
 
 endAnimation = (elem, duration) ->
   if Backbone.$
-    elem.animate {opacity: 1}, duration
+    $(elem).animate {opacity: 1}, duration
```
Comment (akre54) [inline src/chaplin/views/collection_view.coffee]: use `Backbone.$` here instead of just `$`

**Comment by Florian-R** [comment]
All done. I've also changed the call to `$` by `Backbone.$` in `toggleElement` for the sake of consistency.

**Comment by akre54** [comment]
Awesome. Thanks.

**Comment by Florian-R** [comment]
Meh. I see [Travis](https://travis-ci.org/chaplinjs/chaplin/builds/46461791) still fail in the original PR. I must have fucked up something with Grunt when i've rerunned the tests. Won't have time to work on this before Monday, but i'll sent another PR ASAP.

**Pull request merged**